src/miniscript/mod.rs

```diff
@@ -30,7 +30,7 @@ use std::{fmt, str};
 use bitcoin;
 use bitcoin::blockdata::script;
 
-pub use self::context::{BareCtx, Legacy, Segwitv0};
+pub use self::context::{BareCtx, Legacy, Segwitv0, Tap};
```

Comment:
It is really strange to have types named `BareCtx` and `Legacy`/`Tap` in the same time. IMO either should be `Bare`, `Legacy`, `Tap` - or `BareCtx`, `LegacyCtx` and `TapCtx`.

Reply:
Initially everything was without `Ctx`. But `Bare` could have been confused with the `Bare/Raw` descriptor. That is why for `Bare`, I added `BareCtx`. I agree that we should be consisted, but given the limited usage of `Bare`/`BareCtx`, I think saving to type `Ctx` argument wing.